### PR TITLE
clone_job.pl: Skip downloading assets generated by parent jobs

### PR DESCRIPTION
**Pull request by okurz**
In case all dependencies on a job are cloned including a parent job which will
generate an asset which is used by the cloned child there is no need to
download the corresponding asset when it will get overwritten by the new
scheduled parent job anyway.

This might also help to prevent confusion when the option
'--skip-chained-deps' is not used as the qcow image is downloaded but not
used.

Example test run:
```
$ clone_job.pl --from https://openqa 1002                           
Cloning dependencies of foo
downloading                                                    
https://openqa/tests/1002/asset/iso/foo.iso
to                                                             
/var/lib/openqa/factory/iso/foo.iso
Created job #1: bar
downloading
https://openqa/tests/1002/asset/iso/foo.iso
to
/var/lib/openqa/factory/iso/foo.iso
Created job #2: foo
$ clone_job.pl --skip-chained-deps --from https://openqa 1002
downloading
https://openqa/tests/1002/asset/iso/foo.iso
to
/var/lib/openqa/factory/iso/foo.iso
downloading
https://openqa/tests/1002/asset/hdd/baz.qcow2
to
/var/lib/openqa/factory/hdd/baz.qcow2
Created job #3: foo
```

### CODE REVIEW

**Comment by okurz** [comment]
@aaannz as discussed

File: script/clone_job.pl [hunk 1]
```diff
@@ -202,10 +213,16 @@ sub clone_job {
         $job->{settings}->{_START_AFTER_JOBS} = join(',', @new_chained)  if @new_chained;
     }
 
-    for my $type (keys %{$job->{assets}}) {
+  ASSET: for my $type (keys %{$job->{assets}}) {
         next if $type eq 'repo';    # we can't download repos
         for my $file (@{$job->{assets}->{$type}}) {
             my $dst = $file;
+            unless ($options{'skip-deps'} || $options{'skip-chained-deps'}) {
+                my @parents = map { get_job($_) } @{$job->{parents}->{Chained}};
```
Comment (aaannz) [inline script/clone_job.pl]: Is it possible to move this line before the first for loop (i.e. to L215)? The `$job` is not changing so IMO it's wasteful to recalculate `@parents` each iteration.

Comment (okurz) [inline script/clone_job.pl]: changed. Isn't perl smart enough to detect that the objects don't change and use a cached result from memory?

Comment (aaannz) [inline script/clone_job.pl]: I don't know, it very well may be. This one was just obvious enough to bother me. :)

**Comment by aaannz** [comment]
:+1:

**Pull request merged**